src/MDMPerforming.h

```diff
@@ -114,3 +114,31 @@ NS_SWIFT_NAME(ComposablePerforming)
     NS_SWIFT_NAME(set(transactionEmitter:));
 
 @end
+
+/** Specifics for a named plan performer to allow named plans to be added and removed. */
+NS_SWIFT_NAME(NamedPlanPerforming)
+@protocol MDMNamedPlanPerforming <MDMPerforming>
+
+@optional
+
+/**
+ Provides the performer with an plan and a name.
+ 
+ @param plan The plan that required this type of performer.
+ @param name The name by which the plan can be identified.
+ */
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:withName:));
+
+/**
+ Removes a named plan from a performer.
+ 
+ @param plan The plan that required this type of performer.
+ @param name The name by which the plan can be identified.
+ */
+- (void)removePlan:(nonnull id<MDMPlan>)plan
+          withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(remove(plan:withName:));
```

Comment:
I was kinda looking for consistency between the `add` and `remove` APIs. I think its a fair assumption that if the underlying performer is considering using named plans, then it is probably keeping a reference to them indexed on those same names.